src/smart-components/role/roles.js

```diff
@@ -29,40 +31,58 @@ const columns = [
 
 const selector = ({ roleReducer: { roles, isLoading } }) => ({
   roles: roles.data,
-  pagination: roles.meta,
+  meta: roles.meta,
+  filters: roles.filters,
   userIdentity: roles.identity,
   isLoading,
 });
 
 const Roles = () => {
-  const [filterValue, setFilterValue] = useState('');
   const dispatch = useDispatch();
   const { push } = useHistory();
-  const { roles, isLoading, pagination, userIdentity } = useSelector(selector, shallowEqual);
-  const fetchData = (options) => dispatch(fetchRolesWithPolicies(options));
+  const { roles, isLoading, filters, meta, userIdentity } = useSelector(selector, shallowEqual);
+  const fetchData = (options) => dispatch(fetchRolesWithPolicies({ ...options, inModal: false }));
+  const history = useHistory();
+
+  let pagination = roles.pagination || meta;
```

Comment:
We should not be using global variable and then accessing/overriding it in an effect. You risk getting your data out of sync because the variable is overridden on each render. Use `useState` or `useRef` instead.

Reply:
Thank you, moved to useEffect